crates/predictor/src/predictor/point.rs

```diff
@@ -101,16 +107,45 @@ impl Point {
         //TODO: make a fast lookup structure for this
         let levels = [2, 3, 5, 7, 10, 20, 30, 50, 70, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700, 750, 800, 850, 900, 925, 950, 975, 1000];
         let mut best_level : i32 = 1;
+        let mut second_best_level : i32 = 1;
+
         let mut best_level_diff : f32 = (isobaric_hpa - (best_level as f32)).abs();
+        let mut second_best_level_diff : f32 = (isobaric_hpa - (second_best_level as f32)).abs();
 
         for level_ref in levels.iter() {
             let level = *level_ref as i32;
             let diff = (isobaric_hpa - (level as f32)).abs();
 
             if diff < best_level_diff {
```

Comment:
This code won't actually do quite what you want. Imagine levels are [5, 10, 20], and you get an altitude with level 11. `best_level` will be 10, as it should, but `second_best_level` will be 5, when it should actually be 20 for you to be interpolating between the correct things.